### PR TITLE
[RND-300] add acknowledged checks to mongodb writes

### PR DESCRIPTION
**Pull request by bradbanister**

### CODE REVIEW

File: Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Delete.ts [hunk 1]
```diff
@@ -70,8 +70,10 @@ export async function deleteDocumentById(
       // Perform the document delete
       Logger.debug(`mongodb.repository.Delete.deleteDocumentById: Deleting document id ${id}`, traceId);
 
-      const { deletedCount } = await mongoCollection.deleteOne({ _id: id }, { session });
-      deleteResult = deletedCount === 0 ? { response: 'DELETE_FAILURE_NOT_EXISTS' } : { response: 'DELETE_SUCCESS' };
+      const { acknowledged, deletedCount } = await mongoCollection.deleteOne({ _id: id }, { session });
+      if (acknowledged) {
```
Comment (bradbanister) [inline Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Delete.ts:74]: This was the only area I found to be lacking. If `acknowledged` is false, the rest of the write operation results come back as `undefined`, which is not well expressed in the TypeScript types.

File: Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Delete.ts [hunk 2]
```diff
@@ -70,8 +70,10 @@ export async function deleteDocumentById(
...
+        deleteResult = deletedCount === 0 ? { response: 'DELETE_FAILURE_NOT_EXISTS' } : { response: 'DELETE_SUCCESS' };
+      }
```
Comment (stephenfuqua) [inline Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Delete.ts:76]:
```suggestion
      }
      else {
          const msg = 'mongoCollection.deleteOne returned acknowledged = false, which should not be possible in this application';
          Logger.error('mongodb.repository.Delete.deleteDocumentById', traceId, msg);
      }
```

File: Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Update.ts [hunk 3]
```diff
@@ -60,8 +60,10 @@ export async function updateDocumentById(
       // Perform the document update
       Logger.debug(`mongodb.repository.Update.updateDocumentById: Updating document id ${id}`, traceId);
 
-      const { matchedCount } = await mongoCollection.replaceOne({ _id: id }, document, { session });
-      updateResult.response = matchedCount > 0 ? 'UPDATE_SUCCESS' : 'UPDATE_FAILURE_NOT_EXISTS';
+      const { acknowledged, matchedCount } = await mongoCollection.replaceOne({ _id: id }, document, { session });
+      if (acknowledged) {
+        updateResult.response = matchedCount > 0 ? 'UPDATE_SUCCESS' : 'UPDATE_FAILURE_NOT_EXISTS';
+      }
```
Comment (stephenfuqua) [inline Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Update.ts:66]:
```suggestion
      }
      else {
          const msg = 'mongoCollection.replaceOne returned acknowledged = false, which should not be possible in this application';
          Logger.error('mongodb.repository.Update.updateDocumentById', traceId, msg);
      }
```

File: Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Upsert.ts [hunk 4]
```diff
@@ -85,8 +85,10 @@ export async function upsertDocument(
       // Perform the document upsert
       Logger.debug(`mongodb.repository.Upsert.upsertDocument: Upserting document id ${id}`, traceId);
 
-      const { upsertedCount } = await mongoCollection.replaceOne({ _id: id }, document, asUpsert(session));
-      upsertResult.response = upsertedCount === 0 ? 'UPDATE_SUCCESS' : 'INSERT_SUCCESS';
+      const { acknowledged, upsertedCount } = await mongoCollection.replaceOne({ _id: id }, document, asUpsert(session));
+      if (acknowledged) {
+        upsertResult.response = upsertedCount === 0 ? 'UPDATE_SUCCESS' : 'INSERT_SUCCESS';
+      }
```
Comment (stephenfuqua) [inline Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Upsert.ts:91]:
```suggestion
      }
      else {
          const msg = 'mongoCollection.replaceOne returned acknowledged = false, which should not be possible in this application';
          Logger.error('mongodb.repository.Upsert.upsertDocument', traceId, msg);
      }
```

**Comment by bradbanister** [comment]
Incorporated logging error suggestions, added info on why it's a problem.

**Pull request merged**